### DISCUSSION TITLE
On-Prem EMQX Broker Communication with other machine

### DISCUSSION THREAD

**Discussion by AliHaider104** [emqx/emqx#7131] [Q&A] [answered]
Hello Everyone,

I have an EMQX Broker Installed on my machine locally. I have a MQTT client on that machine and I am able to connect that client with my broker. Is it any way I can communicate a MQTT client on an other machine with my locally installed EMQX Broker?

Thanks

**Comment by zmstone** [comment, marked as answer]
If the network is fine.
most likely acl is not configured to allow login.
Enabled debug level logging, you’ll get alot more insights

**Reply by AliHaider104** [reply]
I am using SSL certificates. Connections established successfully when I use MQTT client on same machine where I am running EMQX Broker but when I try to connect MQTT client from an other machine it just show ECONNECREFUSED 127.0.0.1:8883 Error.

**Reply by AliHaider104** [reply]
Here is an update of what I have found. I was running EMQX broker on 1 machine and MQTT client on 2nd but instead of writing the IP address of the 1st machine I was writing the localhost or 127.0.0.1 that was causing the error. The problem is still not completely solved as even after putting the IP address of the machine it is not connecting with Broker.

**Reply by zmstone** [reply]
Hi @AliHaider104 
127.0.0.1 is a loopback address which always points to the local host.
the 2nd machine will try to connect itself if you use 127.0.0.1:8883.

you should use the IP or FQDN of the 1st machine when connecting from 2nd.

**Reply by AliHaider104** [reply]
Yes you are right I have enabled the port 8883 for the 1st VM and also use the Public IP address of that VM and use that instead of 127.0.0.1 Now it's working Thanks!
